compactor/periodic.go

```diff
@@ -73,24 +77,20 @@ func (t *Periodic) Run() {
 					continue
 				}
 			}
-
-			if clock.Now().Sub(last) < executeCompactionInterval {
+			if clock.Now().Sub(last) < t.period {
```

Comment:
Can we simplify the timer logic a bit here?  I found the logic a bit difficult to follow with the `case <-clock.After(checkCompactionInterval):` select case and the `periodDivisor` combined together.

Could we instead do something like 
```
select {
...
case <-clock.After(t.period - clock.Now().Sub(last)):
  ...
```

and then eliminate this `if clock.Now().Sub(last) < t.period {` check and the `periodDivisor`?  

(let me know if I'm missing an important subtlety to how the timers are being use here...)

Reply:
@jpbetz I am trying to minimize any change. we can refactor the code in a future pr if you want.

Reply:
Sounds good.

Reply:
Sounds good.